nuxeo-full-ps-template.yaml

```diff
@@ -0,0 +1,1989 @@
+                      }
+                      stage("Build Nuxeo Package") {
+                          withMaven() {
+                              withCredentials([
+                                  usernamePassword(
+                                      credentialsId: "${project}-${APPLICATION_NAME}-connect-username-password",
+                                      usernameVariable: "NUXEO_CONNECT_USERNAME",
+                                      passwordVariable: "NUXEO_CONNECT_PASSWORD"
+                                  )
+                              ]) {
+                                  unstash name: "mavenSettings"
+                                  def mavenSettings = findFiles(glob: "**/settings.xml")[0]
+                                  sh "mvn --settings ${mavenSettings.path} clean package"
+                                  if (fileExists(".nuxeo-s2i")) {
+                                      def props = readProperties file: ".nuxeo-s2i"
+                                      packagePathRegExp = props["NUXEO_PACKAGE_DIR"]
+                                  } else {
+                                      packagePathRegExp = "*(marketplace|package)-*.zip"
+                                  }
+                                  def files = findFiles(glob: "**/${packagePathRegExp}")
+                                  if (files.length == 0) {
+                                      echo "Marketplace zip file not found."
+                                  }
+                                  stash name:"marketplace", includes:"${files[0]}"
+                              }
+                          }
+                      }
+                    }
+                  node {
+                      stage("Build Image") {
+                          unstash name: "marketplace"
+                          def files = findFiles(glob: "**/${packagePathRegExp}")
+                          if (files.length == 0) {
+                              echo "Marketplace zip file not found."
+                          }
+                          if (fileExists("source")) {
+                              sh "rm -rf source"
+                          }
+                          sh "mkdir -p source"
+                          fileOperations([
+                              fileCopyOperation(
+                                  includes: "${files[0].path}",
+                                  targetLocation: "source",
+                                  flattenFiles: true,
+                              ),
+                              fileRenameOperation(
+                                  source: "source/${files[0].name}",
+                                  destination: "source/marketplace.zip"
+                              ),
+                              fileDownloadOperation(
+                                  url: "https://raw.githubusercontent.com/nuxeo-sandbox/nuxeo-s2i/${NUXEO_VERSION}/contrib/install.sh",
```

Comment:
We should probably find a better way to get this file as it's not in the current repo (which means a dependency that is not obvious here)

Reply:
any good idea?